client/blocks/site-preview/index.jsx

```diff
@@ -1,110 +1,62 @@
+		getSiteOption( state, siteId, 'frame_nonce' )
+	);
+	const previewUrl = useSelector( getPreviewUrl );
+	const hideSEO = useSelector( ( state ) => isDomainOnlySite( state, siteId ) );
 
-	state = {
-		previewCount: 0,
-	};
-
-	previewCounter = 0;
-
-	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		if ( this.props.selectedSiteId && this.props.selectedSiteId !== nextProps.selectedSiteId ) {
-			this.previewCounter = 0;
-		}
-
-		if ( ! this.props.showPreview && nextProps.showPreview ) {
-			debug( 'forcing refresh' );
-			this.previewCounter > 0 && this.setState( { previewCount: this.previewCounter } );
-			this.previewCounter += 1;
-		}
-	}
-
-	getPreviewUrl() {
-		if ( ! this.props.selectedSiteUrl && ! this.props.previewUrl ) {
-			debug( 'no preview url and no site url were found for this site' );
+	function formatPreviewUrl() {
+		if ( ! previewUrl ) {
 			return null;
 		}
-		const previewUrl = addQueryArgs(
+
+		return addQueryArgs(
 			{
 				iframe: true,
 				theme_preview: true,
-				'frame-nonce': this.props.selectedSiteNonce,
-				cachebust: this.state.previewCount,
+				'frame-nonce': selectedSiteNonce ?? '',
+				cachebust: previewCount,
 			},
-			this.getBasePreviewUrl()
+			previewUrl
 		);
-		debug( 'using this preview url', previewUrl );
-		return previewUrl;
 	}
 
-	getBasePreviewUrl() {
-		return this.props.previewUrl || this.props.selectedSiteUrl;
-	}
-
-	render() {
-		if ( ! this.props.selectedSitePreviewable ) {
-			debug( 'a preview is not available for this site' );
-			return null;
-		}
-
-		return (
-			<WebPreview
-				className={ this.props.className }
-				previewUrl={ this.getPreviewUrl() }
-				externalUrl={ this.getBasePreviewUrl() }
-				showExternal={ true }
-				showClose={ true }
-				showPreview={ this.props.showPreview }
-				onClose={ this.props.closePreview }
-				showSEO={ ! this.props.isDomainOnlySite }
-			/>
-		);
-	}
+	return (
+		<WebPreview
+			className={ className }
+			previewUrl={ formatPreviewUrl() }
+			externalUrl={ previewUrl }
+			showExternal={ true }
+			showClose={ true }
+			showPreview={ showPreview }
+			onClose={ () => {
+				dispatch( closePreview() );
+				incrementPreviewCount();
+			} }
+			showSEO={ ! hideSEO }
+		/>
+	);
 }
 
-function mapStateToProps( state ) {
-	const selectedSiteId = getPreviewSiteId( state );
-	// Force https to prevent mixed content errors in the iframe
-	const siteUrl = 'https://' + getSiteSlug( state, selectedSiteId );
+export default function SitePreview( props ) {
+	const siteId = useSelector( getPreviewSiteId );
+	const isPreviewable = useSelector( ( state ) => isSitePreviewable( state, siteId ) );
 
-	return {
-		showPreview: getCurrentLayoutFocus( state ) === 'preview',
-		selectedSite: getPreviewSite( state ),
-		selectedSiteId,
-		selectedSiteUrl: siteUrl.replace( /::/g, '/' ),
```

Comment:
Seems like the support for sites installed in subdirectories is lost. We should keep the `.replace( /::/g, '/' )` logic in our `previewUrl` declaration in order to keep supporting those.

Reply:
Scratch that. I was unable to reproduce it, and it seems like it won't occur anymore because of how we retrieve the preview URL.